model/connect/MySQLiConnector.php

```diff
@@ -235,18 +238,16 @@ public function preparedQuery($sql, $parameters, $errorLevel = E_USER_ERROR) {
 			$statement->execute();
 			return $statement;
 		});
-
-		// check result
-		$this->setLastStatement($lastStatement);
+		
 		if (!$lastStatement || $lastStatement->error) {
 			$values = $this->parameterValues($parameters);
 			$this->databaseError($this->getLastError(), $errorLevel, $sql, $values);
 			return null;
 		}
 
-		// May not return result for non-select statements
-		if($result = $lastStatement->get_result()) {
-			return new MySQLQuery($this, $result, $lastStatement);
+		// Non-select queries will have no result data
+		if($lastStatement && ($metaData = $lastStatement->result_metadata())) {
```

Comment:
No it won't. Only "select" statements return results_metadata. The metadata is the column details of the results. Essentially the metadata of this value is a zero-length resultset containing the same results contained by the statement. It's reasonably safe to assume there is a 1-to-1 correlation between results_metadata and the statement containing results.